internal/command/views/json/diagnostic.go

```diff
@@ -408,6 +412,12 @@ func NewDiagnostic(diag tfdiags.Diagnostic, sources map[string][]byte) *Diagnost
 					diagnostic.Snippet.FunctionCall = callInfo
 				}
 
+				// When the diagnostic is caused by a failed run whose assertion is a binary expression,
+				// we'll include a diff of the two values in the diagnostic snippet.
+				if expr, ok := fromExpr.Expression.(*hclsyntax.BinaryOpExpr); ok && tfdiags.DiagnosticCausedByTestFailure(diag) {
+					diagnostic.Snippet.Values = diffBinaryFailedRunDiagnostic(ctx, expr)
```

Comment:
Right now, this is a replacement, thus removing all existing snippets. It could be that users still want to see the individual traversals, instead of just the LHS and RHS values.